src/components/ItemGrid/ItemGrid.js

```diff
@@ -49,6 +50,15 @@ export class ItemGrid extends Component {
         expandedItems: {},
     };
 
+    constructor(props) {
+        super(props);
+
+        this.getMemoizedItem = memoize(this.getItem);
+        this.getMemoizedItems = memoize(this.getItems);
+        this.getMemoizedItemComponent = memoize(this.getItemComponent);
+        this.getMemoizedItemComponents = memoize(this.getItemComponents);
```

Comment:
I tried using each of these on their own. The only one that actually decreased number of renders was getMemoizedItem (line 56). So I think we can safely remove the other 3 and have the same result.